spec/ic.did

```diff
@@ -16,6 +15,35 @@ type definite_canister_settings = record {
   freezing_threshold : nat;
 };
 
+type canister_change_origin = variant {
+  from_user : record {
+    user_id : principal;
+  };
+  from_canister : record {
+    canister_id : principal;
+    canister_version : opt nat64;
+  };
+};
+
+type canister_change = variant {
+  canister_creation;
+  code_uninstall;
+  code_deployment : record {
+    mode : variant {install; reinstall; upgrade};
+    module_hash : blob;
+  };
+  controllers_change : record {
+    controllers : vec principal;
+  };
+};
+
+type canister_change_entry = record {
+  timestamp : nat64;
```

Comment:
@alin-at-dfinity suggested to call this timestamp_nanos. Agreed?

Reply:
Sounds good.